### PR TITLE
Data processing

### PR DESCRIPTION
**Pull request by Knarik1**
This PR implements the following:

1. Set up a data processing style from the config file
2. Integrate data processing utils from previous Chemlactica project
3. Add a sample for test training in the train_mini folder

### CODE REVIEW

File: torchtitan/datasets/hf_datasets.py [hunk 1]
```diff
@@ -30,6 +33,11 @@
 _supported_datasets = {
     "c4_test": "test/assets/c4_test",
     "c4": "allenai/c4",
+    "train_mini": "test/assets/train_mini"
```
Comment (philippguevorguian) [inline torchtitan/datasets/hf_datasets.py:36]: Let's give it a name like chemlactica_train_mini

File: torchtitan/datasets/hf_datasets.py [hunk 2]
```diff
@@ -102,9 +112,20 @@ def __init__(
         else:
             ds = load_dataset(dataset_path, split="train")
 
+
+        if data_processing_style in _supported_data_processing_styles:
+            data_processing_fn = eval(_supported_data_processing_styles[data_processing_style])
```
Comment (philippguevorguian) [inline torchtitan/datasets/hf_datasets.py:117]: Why do we use eval here?

Comment (philippguevorguian) [reply torchtitan/datasets/hf_datasets.py:117]: maybe something like this:
```python
try:
    data_processing_fn = _supported_data_processing_styles[style]
except KeyError as e:
    raise ValueError(f"Unsupported data processing style: {style}") from e```
    
    Could even make this an enum which maps to a Callable type.

Comment (Knarik1) [reply torchtitan/datasets/hf_datasets.py:117]: Ok, good point, I'll update it.

File: torchtitan/datasets/hf_datasets.py (hunk 1 above)
Comment (Knarik1) [reply torchtitan/datasets/hf_datasets.py:36]: OK

**Comment by Knarik1** [comment]
@philippguevorguian Codes are updated, please look at them one more time.

**Review by philippguevorguian — APPROVED** [review]
LGTM

**Pull request merged**